src/System.Private.ServiceModel/tests/Common/Scenarios/Endpoints.cs

```diff
@@ -53,390 +187,394 @@ public static string HttpProtocolError_Address
+            EnsureLocalClientCertifciateInstalled();
+            return GetEndpointAdddress("WebSocketHttpsDuplexBinaryBuffered.svc//WebSocketHttpsDuplexBinaryBufferedResource", protocol: "wss");
         }
     }
 
     public static string WebSocketHttpsDuplexTextBuffered_Address
     {
         get
         {
-            BridgeClientCertificateManager.InstallRootCertificateFromBridge();
-            return BridgeClient.GetResourceAddress("WcfService.TestResources.WebSocketHttpsDuplexTextBufferedResource");
+            EnsureLocalClientCertifciateInstalled();
+            return GetEndpointAdddress("WebSocketHttpsDuplexTextBuffered.svc//WebSocketHttpsDuplexTextBufferedResource", protocol: "wss");
         }
     }
 
     // net.tcp Addresses
     public static string Tcp_DefaultBinding_Address
     {
-        get { return BridgeClient.GetResourceAddress("WcfService.TestResources.TcpDefaultResource"); }
+        get { return GetEndpointAdddress("TcpDefault.svc//tcp-nosecurity", protocol: "net.tcp"); }
     }
 
     public static string Tcp_NoSecurity_Address
     {
-        get { return BridgeClient.GetResourceAddress("WcfService.TestResources.TcpNoSecurityResource"); }
+        get { return GetEndpointAdddress("TcpNoSecurity.svc//tcp-nosecurity", protocol: "net.tcp"); }
     }
 
     public static string Tcp_Streamed_NoSecurity_Address
     {
-        get { return BridgeClient.GetResourceAddress("WcfService.TestResources.TcpStreamedNoSecurityResource"); }
+        get { return GetEndpointAdddress("TcpStreamedNoSecurity.svc//tcp-streamed-nosecurity", protocol: "net.tcp"); }
     }
 
     public static string Tcp_VerifyDNS_Address
     {
         get
         {
-            BridgeClientCertificateManager.InstallLocalCertificateFromBridge(); 
-            return BridgeClient.GetResourceAddress("WcfService.TestResources.TcpVerifyDNSResource");
+            EnsureLocalClientCertifciateInstalled();
+            return GetEndpointAdddress("TcpVerifyDNS.svc//tcp-VerifyDNS", protocol: "net.tcp");
         }
     }
 
     public static string Tcp_VerifyDNS_HostName
     {
         get
         {
-            return BridgeClient.GetResourceHostName("WcfService.TestResources.TcpVerifyDNSResource");
+            return GetServiceHostName();
         }
     }
 
     public static string Tcp_ExpiredServerCertResource_Address
     {
         get
         {
-            return BridgeClient.GetResourceAddress("WcfService.TestResources.TcpExpiredServerCertResource");
+            EnsureRootCertificateInstalled();
+            return GetEndpointAdddress("TcpExpiredServerCert.svc//tcp-ExpiredServerCert", protocol: "net.tcp");
         }
     }
 
     public static string Tcp_RevokedServerCertResource_HostName
     {
         get
         {
-            return BridgeClient.GetResourceHostName("WcfService.TestResources.TcpRevokedServerCertResource");
+            return GetServiceHostName();
         }
```

Comment:
IIS only has restriction on https port. For TCP port, we can have as many certs as we want.